shakenfist/external_api/app.py

```diff
@@ -608,6 +608,26 @@ def post(self, name=None, cpus=None, memory=None, network=None,
         # We need to sanitise the name so its safe for DNS
         name = re.sub(r'([^a-zA-Z0-9_\-])', '', name)
 
+        # Sanity check
+        if not disk:
+            return error(400, "Instance must specify at least one disk")
+        for d in disk:
+            if not isinstance(d, dict):
+                return error(400, "Disk specification should contain JSON objects")
+            if not all (k in d for k in ('size', 'base', 'bus', 'type')):
+                return error(400, "Disk specification is missing parameters: %s" % d)
```

Comment:
Some of these are optional though...

* size if not set defaults to the size of the base image
* base is only needed for the boot disk
* bus defaults to the configuration item if not set
* type defaults to disk (the other option is CDROM)

Reply:
Got carried away here obviously. Reversing.
I got so many 500 errors I started looking for culprits.